src/reststate-vuex.js

```diff
@@ -323,7 +323,13 @@ const resourceModule = ({ name: resourceName, httpClient }) => {
           const oldRecord = getters.byId({ id: record.id });
 
           // remove old relationships first
-          if (oldRecord && oldRecord.relationships) {
+          // it only makes sense if we temper with the record's relationships
+          if (
+            oldRecord &&
+            oldRecord.relationships &&
+            record &&
+            record.relationships
```

Comment:
could you use the optional chaining operator here?
It could make the code a bit more readable since there are 4 conditions to be checked
```suggestion
          if (
            oldRecord?.relationships &&
            record?.relationships
```